lib/Dialect/Torch/Transforms/RecomposeComplexOps.cpp

```diff
@@ -33,7 +36,14 @@ class RecomposeSliceCopy_ : public OpRewritePattern<AtenCopy_Op> {
     if (!matchPattern(sliceOp.getDim(), m_TorchConstantInt(&dim)))
       return failure();
     int64_t end;
-    if (!matchPattern(sliceOp.getEnd(), m_TorchConstantInt(&end)))
+    if (sliceOp.getEnd().getType().isa<Torch::NoneType>())
+      end = INT64_MAX;
```

Comment:
I've implemented that [here](https://github.com/gpetters94/mlir-npcomp/commit/b0e3c533e7bb7f72fe407e0ea5946fb4b4587c25). It seems to mostly work, but there are some problems with `torch.constant` ops hanging around after they should be gone:

```
/home/gap/work/mlir-npcomp/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir_e2e_test/test_suite/slice_like.py:537:41: error: failed to legalize operation 'torch.constant.int'
        xslice = torch.ops.aten.slice(x, 0, 2, -4, 1)
                                        ^
/home/gap/work/mlir-npcomp/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir_e2e_test/test_suite/slice_like.py:537:41: note: see current operation: %0 = "torch.constant.int"() {value = 0 : i64} : () -> !torch.int
// -----// IR Dump After FinalizingBackendTypeConversion Failed (torch-finalizing-backend-type-conversion) //----- //
func.func @forward(%arg0: tensor<?x?x?xf32>, %arg1: tensor<?x?x?xf32>) -> tensor<?x?x?xf32> {
  %int0 = torch.constant.int 0
  %c0_i64 = arith.constant 0 : i64
  %c1_i64 = arith.constant 1 : i64
  %int2 = torch.constant.int 2
  %c2_i64 = arith.constant 2 : i64
  %int-4 = torch.constant.int -4
  %c-4_i64 = arith.constant -4 : i64
  %c0 = arith.constant 0 : index
  %0 = torch_c.from_builtin_tensor %arg1 : tensor<?x?x?xf32> -> !torch.vtensor<[?,?,?],f32>
  %1 = torch_c.from_builtin_tensor %arg0 : tensor<?x?x?xf32> -> !torch.vtensor<[?,?,?],f32>
  %2 = torch_c.to_builtin_tensor %1 : !torch.vtensor<[?,?,?],f32> -> tensor<?x?x?xf32>
  %3 = torch_c.to_builtin_tensor %0 : !torch.vtensor<[?,?,?],f32> -> tensor<?x?x?xf32>
  %dim = tensor.dim %2, %c0 : tensor<?x?x?xf32>
  %4 = arith.index_cast %dim : index to i64
  %5 = arith.cmpi slt, %4, %c2_i64 : i64
  %6 = arith.select %5, %4, %c2_i64 : i64
  %7 = arith.index_cast %6 : i64 to index
  %8 = arith.addi %4, %c-4_i64 : i64
  %9 = arith.cmpi slt, %8, %c0_i64 : i64
  %10 = arith.select %9, %c0_i64, %8 : i64
  %11 = arith.cmpi sgt, %10, %4 : i64
  %12 = arith.select %11, %4, %10 : i64
  %13 = arith.index_cast %12 : i64 to index
  %14 = arith.cmpi sge, %13, %7 : index
  %15 = arith.select %14, %13, %7 : index
  %16 = arith.subi %15, %7 : index
  %17 = arith.index_cast %16 : index to i64
  %18 = torch_c.from_i64 %17
  %19 = torch_c.to_i64 %18
  %20 = torch.prim.min.int %int2, %18 : !torch.int, !torch.int -> !torch.int
  %21 = torch.aten.neg.int %18 : !torch.int -> !torch.int
  %22 = torch.prim.max.int %20, %21 : !torch.int, !torch.int -> !torch.int
  %23 = torch_c.to_i64 %22
  %24 = torch.aten.lt.int %22, %int0 : !torch.int, !torch.int -> !torch.bool
  %25 = torch.aten.Int.bool %24 : !torch.bool -> !torch.int
  %26 = torch_c.to_i64 %25
  %27 = arith.subi %c1_i64, %26 : i64
  %28 = arith.muli %23, %27 : i64
  %29 = arith.addi %23, %19 : i64
  %30 = arith.muli %29, %26 : i64
  %31 = arith.addi %28, %30 : i64
  %32 = torch_c.from_i64 %31
  %33 = torch_c.to_i64 %32
  %34 = torch.prim.min.int %int-4, %18 : !torch.int, !torch.int -> !torch.int
  %35 = torch.aten.neg.int %18 : !torch.int -> !torch.int
  %36 = torch.prim.max.int %34, %35 : !torch.int, !torch.int -> !torch.int
  %37 = torch_c.to_i64 %36
  %38 = torch.aten.lt.int %36, %int0 : !torch.int, !torch.int -> !torch.bool
  %39 = torch.aten.Int.bool %38 : !torch.bool -> !torch.int
  %40 = torch_c.to_i64 %39
  %41 = arith.subi %c1_i64, %40 : i64
  %42 = arith.muli %37, %41 : i64
  %43 = arith.addi %37, %19 : i64
  %44 = arith.muli %43, %40 : i64
  %45 = arith.addi %42, %44 : i64
  %46 = torch_c.from_i64 %45
  %47 = torch_c.to_i64 %46
  %48 = arith.subi %47, %33 : i64
  %49 = arith.index_cast %48 : i64 to index
  %50 = tensor.empty(%49) : tensor<?xi64>
  %51 = linalg.generic {indexing_maps = [affine_map<(d0) -> (d0)>], iterator_types = ["parallel"]} outs(%50 : tensor<?xi64>) {
  ^bb0(%out: i64):
    %59 = linalg.index 0 : index
    %60 = arith.index_cast %59 : index to i64
    %61 = arith.addi %33, %60 : i64
    linalg.yield %61 : i64
  } -> tensor<?xi64>
  %expanded = tensor.expand_shape %51 [[0, 1]] : tensor<?xi64> into tensor<?x1xi64>
  %52 = tensor.empty(%49) : tensor<?x1xi32>
  %53 = linalg.generic {indexing_maps = [affine_map<(d0, d1) -> (d0, 0)>, affine_map<(d0, d1) -> (d0, d1)>], iterator_types = ["parallel", "parallel"]} ins(%expanded : tensor<?x1xi64>) outs(%52 : tensor<?x1xi32>) {
  ^bb0(%in: i64, %out: i32):
    %59 = arith.trunci %in : i64 to i32
    linalg.yield %59 : i32
  } -> tensor<?x1xi32>
  %54 = torch_c.from_builtin_tensor %53 : tensor<?x1xi32> -> !torch.vtensor<[?,1],si32>
  %55 = torch_c.to_builtin_tensor %54 : !torch.vtensor<[?,1],si32> -> tensor<?x1xi32>
  %56 = tm_tensor.scatter unique_indices(false) ins(%3, %55 : tensor<?x?x?xf32>, tensor<?x1xi32>) outs(%2 : tensor<?x?x?xf32>) {
  ^bb0(%arg2: f32, %arg3: f32):
    tm_tensor.yield %arg2 : f32
  } -> tensor<?x?x?xf32>
  %57 = torch_c.from_builtin_tensor %56 : tensor<?x?x?xf32> -> !torch.vtensor<[?,?,?],f32>
  %58 = torch_c.to_builtin_tensor %57 : !torch.vtensor<[?,?,?],f32> -> tensor<?x?x?xf32>
  return %58 : tensor<?x?x?xf32>
}
```

I'm combing the debug logs to see if I can figure it out, but it looks like it might be an issue in `ConvertTorchToArith`

Reply:
@ramiro050  I suggest we just add the fragile INT64_MAX and add a TODO there for later fixing. Otherwise, I will just be stuck here.